contracts/Vesting.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./DPT.sol";
+import "hardhat/console.sol";
+
+/// @title Vesting
+/// @author Eldar Dryga
+/// @notice this contract allows you to distribute tokens to users within a certain time
+
+contract Vesting is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    IERC20 private immutable _token;
+
+    uint256 public periodOfDistributionInDays;
+    uint256 public amountOfDistributionPerUser;
+    bool public initialized;
```

Comment:
No need in this variable, you can check if u vesting started checking this variable periodOfDistributionInDays is not zero and then when u stop vesting, define as zero this variable